src/components/instantlaunches/launch/index.js

```diff
@@ -29,13 +30,19 @@ import ErrorTypography from "components/error/ErrorTypography";
 import DEErrorDialog from "components/error/DEErrorDialog";
 
 const InstantLaunchStandalone = (props) => {
-    const { id: instant_launch_id, showErrorAnnouncer } = props;
+    const {
+        id: instant_launch_id,
+        resource: resource_path,
+        showErrorAnnouncer,
+    } = props;
     const [config] = useConfig();
     const [userProfile] = useUserProfile();
     const [computeLimitExceeded, setComputeLimitExceeded] = useState(
         !!config?.subscriptions?.enforce
     );
     const [errorDialogOpen, setErrorDialogOpen] = useState(false);
+    const [resource, setResource] = useState(!!resource_path ? null : {});
```

Comment:
Since the `InstantLaunchButtonWrapper` is already using `resource = {}` in its props, this could just be `useState(null)`.

Reply:
I think that doesn't work, from what I could find? The destructuring assignment defaults do that with `undefined` but not with `null` according to MDN ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_value )

![image](https://github.com/user-attachments/assets/44edb5f5-6f50-423c-875e-e62114bd5b2d)

(and I figured that sticking `undefined` in react state would probably explode somehow, lol)

Reply:
Looking deeper, though, we're just checking the truthiness of `resource` later on, so null would work fine. I'll change it.

Reply:
Oops, you're right about the destructuring, but there is no problem using `undefined` with `useState`. We use `useState()` all over the place.

I think in Typescript, depending on how your types are defined, it might complain if you use `undefined` with `useState` instead of an empty string or something else, but we haven't incorporated any Typescript in our project yet 😅